service/api_task.go

```diff
@@ -312,6 +311,49 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(w, endTaskResp)
 }
 
+// prepareForReprovision readies host for reprovisioning.
+func prepareForReprovision(ctx context.Context, env evergreen.Environment, settings *evergreen.Settings, h *host.Host, w http.ResponseWriter) error {
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	if err := h.StopAgentMonitor(ctx, env); err != nil {
```

Comment:
This is potentially racy, since we aren't guaranteeing that the agent has cleanly exited before stopping the agent monitor. Are you sure this won't cause problems?

Reply:
The agent and the monitor are completely separate processes, so killing the agent monitor only prevents the agent from respawning. If we assume that the agent monitor is stopped before the agent, then it ensures that both have stopped by the time we start switching host provisioning.